include/boost/iterator/filter_iterator.hpp

```diff
@@ -62,20 +56,20 @@ namespace iterators {
       filter_iterator() { }
 
       filter_iterator(Predicate f, Iterator x, Iterator end_ = Iterator())
-          : super_t(BOOST_ITERATOR_DETAIL_MOVE(Iterator, x)), m_predicate(BOOST_ITERATOR_DETAIL_MOVE(Predicate, f)), m_end(BOOST_ITERATOR_DETAIL_MOVE(Iterator, end_))
+          : super_t(std::move(x)), m_predicate(std::move(f)), m_end(std::move(end_))
```

Comment:
Use `static_cast`s instead of `std::move` to avoid having to include `<utility>` (which is now missing anyway).